SmartPlaces.Facilities/samples/Telemetry.Mapped/src/Processors/TelemetryIngestionProcessor.cs

```diff
@@ -318,47 +323,65 @@ private dynamic TranslateType(TypedValue sourceData, DTEntityKind targetType)
                         case DTEntityKind.String:
                             return sourceData.EnumValue.ToString();
                     }
+
                     break;
                 case TypedValue.ValueOneofCase.Float32Value:
                     switch (targetType)
                     {
                         case DTEntityKind.Double:
-                            return Convert.ToDouble(sourceData.Float32Value);
+                            if (sourceData.Float32Value.ToString() != "NaN")
+                            {
+                                return Convert.ToDouble(sourceData.Float32Value);
+                            }
+
+                            logger.LogError("Received NaN as Float32Value");
```

Comment:
By not throwing here the switch statement falls through to the "Uncertain how to cast from Float32Value to Double" NotImplemented exception log and metric which seems misleading.